src/stacked_borrows/mod.rs

```diff
@@ -806,6 +754,8 @@ trait EvalContextPrivExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             Ok(())
         };
 
+        // FIXME: We create 3 different current_span handles, but a single call to this function
+        // should only create one of the 3 uhhhh awkward?
```

Comment:
Yeah, what's up with that? Doing this at the top of the function should work as before
```rust
let current_span = &mut this.machine.current_span({*this.tcx});
```
not sure if the curlcy braces are truly needed; they force a copy.